### PR TITLE
[KED-2186] Fade graph opacity when loading

### PR DESCRIPTION
**Pull request by richardwestenra**
## Description

I've noticed that often, when loading the graph while changing the pipeline, it's not sufficiently obvious that the chart is still loading. This is because the sidebar updates before the graph does, which gives the impression that it's ready when it isn't yet. This change aims to prevent this by fading out the graph a bit when loading, to make it more obvious that it's still displaying old data.
loading-fade

## Development notes

I haven't added a transition to this, because it wasn't very performant and I wanted it to be snappy

## QA notes

Best observed when changing pipelines, ideally with large pipelines.

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

## Legal notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Review by bru5 — CHANGES_REQUESTED** [review]
I tried this out and when the loading is quick, for example toggling tags on the demo, the result is that all nodes appear to flicker every time.

So I think there needs to be a delay before this appears e.g. 500ms and it needs to show for at least e.g. 300ms to avoid too much flicker.

The same idea could also be applied to the loading spinner, though it is less of a problem there.

As for the effect, I think we can do something less obtrusive but I agree with the reasoning behind it.

File: src/components/flowchart/index.js [hunk 1]
```diff
@@ -433,12 +434,14 @@ export class FlowChart extends Component {
...
    */
   render() {
-    const { chartSize, layers } = this.props;
+    const { chartSize, isLoading, layers } = this.props;
     const { outerWidth = 0, outerHeight = 0 } = chartSize;
 
     return (
       <div
-        className="pipeline-flowchart kedro"
+        className={classnames('pipeline-flowchart kedro', {
```
Comment (bru5) [inline src/components/flowchart/index.js:442]: Every time this class is added then removed, there is potential for creating a CPU spike, which will create jank for the graph animation, so I'd suggest to instead apply the opacity directly to a SVG `g` element instead (such as the zoom wrapper).

Comment (richardwestenra) [reply src/components/flowchart/index.js]: Ok cool, I've moved it to the zoom wrapper

**Comment by richardwestenra** [comment]
Closing in favour of #296: After a discussion with @bru5, we feel that when switching pipeline, hiding the graph completely (via a state reset) will accomplish the same effect with fewer negative side-effects.

**Pull request closed without merging**